### ISSUE TITLE
iOS Simulator support for M1 Mac

### ISSUE THREAD

**Issue by timlyeee** [cocos/cocos-engine#10663]
Right now if one would like to build one Xcode project that supports the M1 iOS Simulator, he should modify the CMakelist.txt in the external folder, which means that one cannot use the same project for building the M1 iOS Simulator App and Device App.

What produces this problem is that M1 Mac requires a different static lib for iOS Simulator(arm64), and lipo cannot create a fat lib for both arm64(device) and arm64(simulator). 

We do want to use the xcframework to support it, but the newest version of CMake is still working on it: https://gitlab.kitware.com/cmake/cmake/-/issues/21752

One reason is that `xcframework` is basically a folder containing different static-libs, so CMake should still target the library by using `-f` flag.

## Why should we use different static library for arm64(device) and arm64(simulator)? 

https://developer.apple.com/forums/thread/673387
So basically, Apple is starting to treat iOS Simulator and iOS as two different platforms.

**Comment by timlyeee** [comment]
Even more, [Mac Catalyst](https://developer.apple.com/mac-catalyst/) requires a different static lib(arm64).
